data_model/src/transaction.rs

```diff
@@ -122,6 +129,24 @@ pub trait Txn {
         let tx_timestamp = Duration::from_millis(self.payload().creation_time);
         tx_timestamp.saturating_sub(crate::current_time()) > threshold
     }
+
+    /// Create expired pipeline event for the given transaction.
+    #[cfg(feature = "std")]
+    fn expired_event(&self) -> Event
```

Comment:
While I understand that this was the easiest way to somehow incapsulate the logic of creating *expired event* related to transaction, I wonder, if it may be usefull to introduce strong-typing for expired transaction?

It may look like this:

```rust
impl Transaction {
    pub fn mark_as_expired(self) -> ExpiredTransaction {
        // ...
    }
}

pub struct ExpiredTransaction(Transaction);

/// I'm not a huge fan of this `From for Event` approach, but we already use this a lot
impl From<ExpiredTransaction> for Event {
    // ...
}
```

Or may be even better in future:

```rust
pub struct Transaction<State> {
    // ...
}

enum State {
    Raw,
    Validated,
    Committed,
    Rejected,
    Expired,
}
```

Just a suggestion. Not sure it will be useful

Reply:
I would like to try later approach you proposed one day, but `Expired` state feel like overkill to me, because only use we have for `Expired` transactions is to convert then into `Event`.
However i would like to hear what other reviewers think.

Reply:
I agree. After discussion we have we conclude that we don't need `Expired` transactions in `Block`. So it's okay for me to leave as is